src/Host/Client/Impl/rtvs/R/overrides.R

```diff
@@ -54,6 +54,20 @@ show_file <- function(files, header, title, delete.file) {
     }
 }
 
+
+# Override for edit(...). Opens file in VS editor and blocks until the file is closed.
+edit_file <- function(name, file, title) {
+    if(!is.null(name) && is.function(name)) {
+        source <- send_request_and_get_response("?EditFile", paste0(deparse(name), collapse = '\n'), NULL);
+        parsed <- try(eval.parent(parse(text = source)));
+        return(parsed);
+    } else if(!is.null(file)) {
+        send_request_and_get_response("?EditFile", NULL, file);
+    } else {
+        utils:::edit(name, file, title);
```

Comment:
In fact, this function should only be called on objects that are functions. On everything else, `edit()` just goes directly to the built-in editor. Here's what `edit.default` looks like:
```r
function (name = NULL, file = "", title = NULL, editor = getOption("editor"), ...) 
{
    if (is.null(title)) 
        title <- deparse(substitute(name))
    if (is.function(editor)) 
        invisible(editor(name = name, file = file, title = title))
    else .External2(C_edit, name, file, title, editor)
}
```